terminate.js

```diff
@@ -3,20 +3,48 @@ var handlePsTreeCallback = require('./handlePsTreeCallback');
 
 /**
  * terminate is an ultra-simple way to kill all the node processes
- * by providing a process.pid. It finds all child processes and shuts
+ * by providing a process pid. It finds all child processes and shuts
  * them down too, so you don't have to worry about lingering processes.
  * @param {int} pid - the Process ID you want to terminate
- * @param {function} callback (optional) - if you want to know once the
+ * @param {string=SIGTERM} signal (optional) - the signal to kill the processes
+ * with.
+ * @param {object={}} options
+ * @param {number=500} options.pollInterval - interval to poll whether pid
+ * and all of the child pids have been killed.
+ * @param {number=5000} options.timeout - max time to wait for processes to
+ * exit before timing out and calling back with an error.
+ * @param {function=} callback (optional) - if you want to know once the
  * procesess have been terminated, supply a callback.
- *   @param {string} error - will be null if no error occured
- *   @param {boolean} done - will be true if the operation succeeded.
+ *   @param {Error} error - will be null if no error occured
  */
-
-module.exports = function terminate(pid, callback) {
+module.exports = function terminate(pid) {
```

Comment:
@jedwards1211 there _appears_ to be only **one** parameter to the `terminate` function.
and yet there a whole bunch of _optional_ params in the Documentation. 

While I _personally_ (_really_) like the idea of using the [`arguments`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments) Object and making all other arguments to the function _optional_, _some_ people prefer to keep all parameters _explicit_...

Also, for _clarity_, the _incrementing_ of the index should be done `after` the value/object/function has been extracted from `arguments`.

The _tests_ appear to work as _expected_: https://travis-ci.org/dwyl/terminate/jobs/190886264#L427
including the tests which pass in the `options` parameter. 

If someone `else` ever decides they have the time to ***refactor*** the code to use https://github.com/dwyl/goodparts as the "_style guide_" that would be a (_very_) welcome PR... see: https://github.com/dwyl/terminate/issues/29